runtime/src/registrar.rs

```diff
@@ -0,0 +1,381 @@
+		Ok(())
+	}
+}
+
+type BalanceOf<T> =
+	<<T as Trait>::Currency as Currency<<T as system::Trait>::AccountId>>::Balance;
+type NegativeImbalanceOf<T> =
+	<<T as Trait>::Currency as Currency<<T as system::Trait>::AccountId>>::NegativeImbalance;
+
+pub trait Trait: parachains::Trait {
+	/// The overarching event type.
+	type Event: From<Event> + Into<<Self as system::Trait>::Event>;
+
+	/// The system's currency for parathread payment.
+	type Currency: Currency<Self::AccountId>;
+
+	/// The deposit to be paid to run a parathread.
+	type ParathreadDeposit: Get<BalanceOf<Self>>;
+
+	/// Handler for when two ParaIds are swapped.
+	type OnSwap: OnSwap;
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as Registrar {
+		// Ordered vector of all parachain IDs.
+		Parachains: Vec<ParaId>;
+
+		/// The number of threads to schedule per block.
+		ThreadCount: u32;
+
+		/// Current set of threads the next block; ordered by para ID. There can be no duplicates
+		/// of para ID in the list.
+		SelectedThreads: Vec<(ParaId, CollatorId)>;
+
+		/// Parathreads/chains scheduled for execution this block. If the collator ID is set, then
+		/// a particular collator has already been chosen for the next block, and no other collator
+		/// may provide the block.
+		///
+		/// Ordered as Parachains ++ Selected_Parathreads.
+		Active: Vec<(ParaId, Option<CollatorId>)>;
+
+		/// The next unused ParaId value. Start this high in order to keep low numbers for
+		/// system-level chains.
+		NextFreeId: ParaId = LOWEST_USER_ID;
+
+		/// Pending swap operations.
+		PendingSwap: map ParaId => ParaId;
+
+		// Map of all registered parathreads/chains.
+		Paras get(paras): map ParaId => Option<ParaInfo>;
+	}
+	add_extra_genesis {
+		config(parachains): Vec<(ParaId, Vec<u8>, Vec<u8>)>;
+		config(_phdata): PhantomData<T>;
+		build(|storage: &mut StorageOverlay, _: &mut ChildrenStorageOverlay, config: &GenesisConfig<T>| {
+			use sr_primitives::traits::Zero;
+
+			let mut p = config.parachains.clone();
+			p.sort_unstable_by_key(|&(ref id, _, _)| *id);
+			p.dedup_by_key(|&mut (ref id, _, _)| *id);
+
+			let only_ids: Vec<_> = p.iter().map(|&(ref id, _, _)| id).cloned().collect();
+
+			<Parachains as generator::StorageValue<_>>::put(&only_ids, storage);
+
+			for (id, code, genesis) in p {
+				// no ingress -- a chain cannot be routed to until it is live.
+				<parachains::Code as generator::StorageMap<_, _>>
+					::insert(&id, &code, storage);
+				<parachains::Heads as generator::StorageMap<_, _>>
+					::insert(&id, &genesis, storage);
+				<parachains::Watermarks<T> as generator::StorageMap<_, _>>
+					::insert(&id, &Zero::zero(), storage);
+			}
+		});
+	}
+}
+
+decl_module! {
+	/// Parachains module.
+	pub struct Module<T: Trait> for enum Call where origin: <T as system::Trait>::Origin {
+		/// Register a parachain with given code.
+		/// Fails if given ID is already used.
+		#[weight = SimpleDispatchInfo::FixedOperational(5_000_000)]
+		pub fn register_para(origin,
```

Comment:
The module's [`register_para` public function](https://github.com/paritytech/polkadot/pull/341/files#diff-6ceafc4b4dd2600a9e5caebc167c0cecR181) here has `info` as the last argument, whereas the `register_para` traits https://github.com/paritytech/polkadot/pull/341/files#diff-6ceafc4b4dd2600a9e5caebc167c0cecR181 and https://github.com/paritytech/polkadot/pull/341/files#diff-6ceafc4b4dd2600a9e5caebc167c0cecR65 have it as the 2nd argument. Would it add any value to have the argument ordering consistent?